### ISSUE TITLE
[Core] StakLayout Padding update issue

### ISSUE THREAD

**Issue by myroot** [xamarin/Xamarin.Forms#2763]
### Description
 The `Padding` of the `StackLayout` causes a change in the stack size but changes of `Padding` does not trigger MeasureChanged. So Size of StackLayout does not update until re-layouting

* `Padding` affects the `StackLayout` size calculation.
https://github.com/xamarin/Xamarin.Forms/blob/882bf07917bea57b93d0d9becd768dcab739c26f/Xamarin.Forms.Core/StackLayout.cs#L71-L83

* But MeasureChanged does not trigger
https://github.com/xamarin/Xamarin.Forms/blob/882bf07917bea57b93d0d9becd768dcab739c26f/Xamarin.Forms.Core/Layout.cs#L103-L106

Need to override `IPaddingElement.OnPaddingPropertyChanged` on StackLayout like below
``` C#
		void IPaddingElement.OnPaddingPropertyChanged(Thickness oldValue, Thickness newValue)
		{
			InvalidateMeasureInternal(InvalidationTrigger.MeasureChanged);
		}
```

### Steps to Reproduce
 1. Update `Padding` of `StackLayout`

### Expected Behavior
 StackLayout size should be changed and update looks
### Actual Behavior
 Does not update looks

### Basic Information
- Version with issue: master of github
- Last known good version: No

### Screenshots

![image](https://user-images.githubusercontent.com/1029155/40215801-afb877c0-5a9d-11e8-8b1b-f5f4206d54a6.gif)

**Comment by jassmith** [comment]
wow yeah that is obviously wrong

**Comment by andreinitescu** [comment]
Can we use this bug as an opportunity to look to all other controls which use Padding and any other properties (Margin, BorderThickiness, etc.) which must trigger resize?

**Comment by andreinitescu** [comment]
I still wonder why wasn't Padding added to View, just how Margin is.

**Comment by jassmith** [comment]
Because padding is a relationship between a view and its children. Not all views have children so padding wouldn't have any logical meaning there. You could argue that it should just produce a blank space around the control where the background color still fills, but that would require us to do nesting, which we are explicitly trying to get away from.

**Comment by PaulVrugt** [comment]
The fix for this issue (#2764) also fixes some issues we're struggling with in regards to iPhone X support. We would really like to see this fix released
